src/silx/gui/data/DataViews.py

```diff
@@ -1084,7 +1110,17 @@ def normalizeData(self, data):
 
     def setData(self, data):
         data = self.normalizeData(data)
-        self.getWidget().addImage(
+        plot = self.getWidget()
+        imageItem = plot._getItem("image")
+    
+        if imageItem is None:
+            imageItem = ImageDataAggregated()
+            imageItem.setAggregationMode(self.getAggregationModeAction().getAggregationMode())            
+            imageItem.setName("data")
+            imageItem.setColormap(plot.getDefaultColormap())
+            plot.addItem(imageItem)
+
+        plot.addImage(
             legend="data", data=data, resetzoom=self.__resetZoomNextTime
         )
```

Comment:
It sounds best to not call `addImage` here and instead call `ImageDataAggregated.setData` directly + `PlotWidget.resetZoom`

```suggestion
        imageItem.setData(data)
        if self.__resetZoomNextTime:
            plot.resetZoom()
```